client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx

```diff
@@ -20,27 +20,80 @@ export default class RelatedProducts extends React.Component {
       yourOutfitItems:[],
       allPropsObj: [],
       outfitPropsObj: [],
-      rpLoaded: false,
-      yoLoaded: false
+      // rpLoaded: true,
+      // yoLoaded: true,
+      // loaded:true,
 
     }
     this.handleAddToOutfit = this.handleAddToOutfit.bind(this);
     this.handleRemoveFromOutfit = this.handleRemoveFromOutfit.bind(this);
     this.getRelatedStateData = this.getRelatedStateData.bind(this);
     this.getOutfitData = this.getOutfitData.bind(this);
+    this.product = this.product.bind(this)
+    this.style = this.style.bind(this)
+    this.outFit = this.outFit.bind(this)
   }
 
   componentDidMount () {
+    //(products request)
+    let product = this.product()
 
-    this.getRelatedStateData(),
-    this.getOutfitData()
+    //(styles request)
+    let getStyle = this.style()
 
+    //(outfit request)
+    let outFitData = this.outFit()
+    // let outFitData = new Promise((resolve, reject) => {
+    //   axios.get(api + `products/${this.props.state.product_id}/styles`, {
+    //     headers: {
+    //       'Authorization': TOKEN
+    //     }
+    //   })
+    //   .then((styleData)=> {
+    //     let outfitPropsObj = helper.compileYourOutfitDataToProps(this.props.state.productInformation , styleData.data);
+    //     resolve(outfitPropsObj);
+    //   })
+    // })
+    //gather the promises for everything in a single then
+    product.then(data => {
+      getStyle.then(styleData => {
+        outFitData.then(fitData => {
+          // //use your helper to get correct style data
+
+          let resultStyleWithId=[];
+          data.forEach((product, pi) => {
+            styleData.forEach((style, si) => {
+              if (pi === si) {
+              resultStyleWithId.push(helper.addIdToStylesData(style, product.id))
```

Comment:
Think this may be where my helpers are bugging out. I'm wondering if the double loop is adding Id slightly differently than it was prior. Will have to check it out with newfound testing capabilities

Reply:
i think this issue has been resolved since yesterday!  excited to see changes with new pull request you throw up @callumreid